frontend/model/contracts/shared/time.js

```diff
@@ -30,19 +33,27 @@ export function periodStampsForDate (
   let previous, current, next
   if (knownSortedStamps.length) {
     const latest = knownSortedStamps[knownSortedStamps.length - 1]
+    const earliest = knownSortedStamps[0]
     if (timestamp >= latest) {
       current = periodStampGivenDate({ recentDate: timestamp, periodStart: latest, periodLength })
       next = plusOnePeriodLength(current, periodLength)
       previous = current > latest
         ? minusOnePeriodLength(current, periodLength)
         : knownSortedStamps[knownSortedStamps.length - 2]
+    } else if (guess && timestamp < earliest) {
+      // handle dates earlier than all known stamps
+      current = periodStampGivenDate({ recentDate: timestamp, periodStart: earliest, periodLength })
+      next = plusOnePeriodLength(current, periodLength)
+      previous = minusOnePeriodLength(current, periodLength)
```

Comment:
Note: this ended up not being needed to close the issue, but I left it because it seems potentially useful to be able to return guessed period stamps based on the period length instead of `undefined`